cop_prinstrat/code/CoP_Moderna.R

```diff
@@ -0,0 +1,132 @@
+#.libPaths()))
+#
+#
+#
+##----------------------------------------------- 
+## obligatory to append to the top of each script
+##renv::activate(project = here::here(".."))
+#renv::activate(project = here::here())
+#
+#
+## There is a bug on Windows that prevents renv from working properly. The following code provides a workaround:
+#if (.Platform$OS.type == "windows") .libPaths(c("C:/Users/yhuang/Documents/renv/library/R-4.0/x86_64-w64-mingw32", "C:/Users/yhuang/AppData/Local/Temp/RtmpUPdelA/renv-system-library",
+#.libPaths()))
+#
+#
+##dat.mock<-read.csv(here::here("data_clean", "practice_data.csv")) 
+index1<-as.integer(commandArgs(trailingOnly=T)[1])
+index2<-as.integer(commandArgs(trailingOnly=T)[2])
+
+library(kyotil)
+library(splines)
+library(nnet)
+
+setwd("~/correlates_reporting")
+
+#dat.mock<-read.csv(here::here("data_clean", "moderna_mock_data_processed.csv")) 
+dat.mock<-read.csv(file="data_clean/moderna_mock_data_processed.csv")
+#dat.mockJ<-read.csv(file="data_clean/janssen_pooled_mock_data_processed.csv")
+    
+      
+assay.seq<-c("bindSpike","bindRBD","bindN","pseudoneutid50","pseudoneutid80","liveneutmn50")
+pop.seq<-c('57','29')
+
+pop=pop.seq[index1]
+y.v<-"EventIndPrimaryD"%.%pop       
+
+ps<-c("Day"%.%pop%.%assay.seq[index2])
+
+if (pop=="57") {
+    dat.mock$wt.0=dat.mock$wt.D57
+    dat.mock$TwophasesampInd.0 = dat.mock$TwophasesampIndD57
+    dat.mock$ph1=dat.mock$ph1.D57   
+} else if (pop=="29") {
+    dat.mock$wt.0=dat.mock$wt.D29
+    dat.mock$TwophasesampInd.0 = dat.mock$TwophasesampIndD29 
+    dat.mock$ph1=dat.mock$ph1.D29
+} else stop("wrong pop")
+
+     
+dat.wide=subset(dat.mock, Bserostatus==0 & !is.na(wt.0))    
+     
+#
+#source(here::here("cop_prinstrat","YingFunctions","NonparFun_NoW.R"))
+#
+#source(here::here("cop_prinstrat","YingFunctions","Fun_2020.R"))
+#
+#source(here::here("cop_prinstrat","YingFunctions","function10_15_Full.R"))
+#
+#source(here::here("cop_prinstrat","YingFunctions","FunctionCall.R"))
+#
+
+source(file="cop_prinstrat/YingFunctions/NonparFun_NoW.R")
+
+source(file="cop_prinstrat/YingFunctions/Fun_COVID.R")
+
+source(file="cop_prinstrat/YingFunctions/function10_15_Full.R")
+
+source(file="cop_prinstrat/YingFunctions/FunctionCall.R")
+
+
+dat_proc<-dat.wide
+study_name_code<-"COVE"
+
+tps.stratums=sort(unique(dat_proc$tps.stratum)); names(tps.stratums)=tps.stratums
+decode.tps.stratum=t(sapply(tps.stratums, function(i) unlist(subset(dat_proc, tps.stratum==i)[1,
+    if (study_name_code=="COVE") c("Senior", "HighRiskInd", "URMforsubcohortsampling") else if (study_name_code=="ENSEMBLE") c("Senior", "HighRiskInd", "Region", "URMforsubcohortsampling")
+])))
+
+
+
+#dat.wide$Agegr<-as.numeric(dat.wide$Age<65)
+
+S1<-dat.wide[,ps]
+Z<-dat.wide$Trt
+Y<-dat.wide[,y.v]
+X<-rep(0,length(Y))
+Xu<-unique(X)
+
+### Since MinorityInd does not have missing value, we use it instead of URMforsubcohortsampling
+W<-dat.wide$MinorityInd*100+dat.wide$HighRiskInd*10+dat.wide$Senior
```

Comment:
I don't really understand what's happening here, but I assume it's appropriate :)